src/components/ValidationStatus.vue

```diff
@@ -118,93 +82,41 @@ export default {
       },
     },
     errorList() {
-      return Object.entries(this.validationErrors).reduce((errorList, [ errorKey, errors ]) => {
-        const errorListItems = errors.map((error) => {
-          return  {
-            id: error.id,
-            errorKey,
-            category: error.category,
-            message: error.message,
-          };
+      return Object.entries(this.validationErrors)
+        .flatMap(([errorKey, errors]) => {
+          return errors.flatMap(error => {
+            return { ...error, errorKey };
+          });
         });
-
-        errorList.push(...errorListItems);
-        return errorList;
-      }, []);
-    },
-    getCategory() {
-      const errorCategory = this.errorList.find((category) => {
-        return category;
-      });
-      return errorCategory;
-    },
-    ellipsisIcon() {
-      return faEllipsisV;
-    },
-    statusIcon() {
-      return this.hasValidationErrors
-        ? faTimesCircle
-        : faCheckCircle;
-    },
-    valditionIcon() {
-      return this.isErrorOrWarning
-        ? faTimesCircle
-        : faExclamationTriangle;
     },
-    isErrorOrWarning() {
-      return this.getCategory.category === 'error' ? true : false;
+    numberOfProblems() {
+      return this.errorList.length + this.warnings.length;
     },
-    iconColor() {
-      return this.isErrorOrWarning
-        ? `${ this.errorColor }`
-        : `${ this.warningColor }`;
-    },
-    statusColor() {
-      return this.hasValidationErrors
-        ? `${ this.errorColor }`
-        : `${ this.validColor }`;
-    },
-    hasValidationErrors() {
-      return this.numberOfValidationErrors > 0;
-    },
-    numberOfValidationErrors() {
-      return this.errorList.length;
+  },
+  methods: {
+    isErrorCategory(error) {
+      return error.category === 'error';
     },
   },
 };
 </script>
 
 <style scoped lang="scss">
 $primary-white: #f7f7f7;
-$seconadry-grey: #555555;
-$secondary-blue: #3397e1;
+$secondary-grey: #555555;
 $border-color: rgba(0, 0, 0, 0.125);
-$id-container-width: 6.5rem;
 $message-container-width: 18rem;
-$error-category-width: 1rem;
 $validation-container-height: 14rem;
 $validation-container-width: 25rem;
 $status-bar-container-height: 3rem;
-$error-color: #D9534F;
-$warning-color: #F0AD4E;
-$button-color: #3BD7FF;
 
 .status-bar-container {
-  color: $seconadry-grey;
+  color: $secondary-grey;
   height: $status-bar-container-height;
-  cursor: pointer;
 
-  &__status {
+  &-status {
```

Comment:
I changed these styles to be dashes instead if underscores as I find the class names a little easier to read. Just a matter of personal preference and I'm happy to change them back if people feel strongly.

Reply:
I'm fine with either or. I usually follow the BEM methodology when naming my classes just personal preference. :)

http://getbem.com/

Reply:
I'm not a css pro, but the existing style is one I see more often in other projects.

Reply:
OK, I'll switch it back if it's the convention.